app/dockerdwrapper.c

```diff
@@ -658,6 +665,44 @@ main(void)
   openlog(NULL, LOG_PID, LOG_USER);
   syslog(LOG_INFO, "Started logging.");
 
+ // Get UID of the current user
+  uid_t uid = getuid();
+
+  char path[strlen(APP_DIRECTORY) + 256];
+  sprintf(path, "/bin:/usr/bin:%s:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin", APP_DIRECTORY);
+
+  char docker_host[256];
+  sprintf(docker_host, "unix://run/user/%d/docker.sock", (int)uid);
+
+  char xdg_runtime_dir[256];
+  sprintf(xdg_runtime_dir, "/run/user/%d", (int)uid);
+
+  // Set environment variables
+  if (setenv("PATH", path, 1) != 0) {
+      perror("Error setting environment PATH");
+      return 1;
```

Comment:
why not use syslog(LOG_ERR,...) and -1 as return code here?
(same comment for the following variables)